contracts/zero-ex/contracts/src/features/MetaTransactionsFeature.sol

```diff
@@ -347,17 +348,34 @@ contract MetaTransactionsFeature is
                     state.mtx.value
                 ).rrevert();
         }
-        // Must be signed by signer.
-        try
-            ISignatureValidatorFeature(address(this))
-                .validateHashSignature(state.hash, state.mtx.signer, state.signature)
-        {}
-        catch (bytes memory err) {
+
+        if (state.signature.length != 66) {
```

Comment:
I do think `state.signature` should be the deconsructed signature already and we unpack it in `executeMetaTransaction()` and `_executeMetaTransaction()`. Maybe `_executeMetaTransactionPrivate()` should just take an `ExecuteState` object as the sole input as well.